src/bgnbd_CET.cpp

```diff
@@ -0,0 +1,153 @@
+
+  return term1 % term2 / term3;
+}
+
+//' @title BG/NBD: Conditional Expected Transactions without covariates
+//'
+//' @description
+//' Calculates the expected number of transactions in a given time period based
+//' on a customer's past transaction behavior and the BG/NBD model parameters.
+//'
+//' @param r TODO: describe
+//' @param alpha TODO: describe
+//' @param a TODO: describe
+//' @param b TODO: describe
+//' @param nPeriods time prediction time frame
+//' @template template_params_rcppxtxtcal
+//'
+//' @details
+//' \code{r} \code{alpha} \code{a} \code{b} estimated parameters in original scale
+//' for the BG/NBD model, namely (r, alpha, a, b).
+//' r and alpha: TODO: description.
+//' a and b: TODO: description
+//'
+//'
+//'@return
+//' Returns a vector containing the conditional expected transactions for the existing
+//' customers in the BG/NBD model.
+//'
+//' @name bgnbd_CET
+// [[Rcpp::export]]
+arma::vec bgnbd_nocov_CET(const double r,
+                    const double alpha,
+                    const double a,
+                    const double b,
+                    const double nPeriods,
+                    const arma::vec& vX,
+                    const arma::vec& vT_x,
+                    const arma::vec& vT_cal){
+
+  // Build alpha and beta --------------------------------------------------------
+  //    No covariates: Same alphas, betas for every customer
+  const double n = vX.n_elem;
+
+  arma::vec vAlpha_i(n), vA_i(n), vB_i(n);
+
+  vAlpha_i.fill(alpha);
+  vA_i.fill(a);
+  vB_i.fill(b);
+
+  return bgnbd_CET(r, vAlpha_i, vA_i, vB_i, nPeriods, vX, vT_x, vT_cal);
+}
+
+//' @title BG/NBD: Conditional Expected Transactions with static covariates
+//'
+//' @description
+//' Calculates the expected number of transactions in a given time period based
+//' on a customer's past transaction behavior and the BG/NBD model parameters.
+//'
+//' @param r TODO: describe
+//' @param alpha TODO: describe
+//' @param a TODO: describe
+//' @param b TODO: describe
+//' @param nPeriods time prediction time frame
+//' @template template_params_rcppxtxtcal
+//' @template template_params_rcppcovmatrix
+//' @template template_params_rcppvcovparams
+//'
+//' @details
+//' \code{r} r: TODO
+//'
+//' \code{r} alpha: TODO
+//'
+//' \code{r} a: TODO
+//'
+//' \code{r} b: TODO
+//'
+//' \code{mCov_trans} is a matrix containing the covariates data of
+//' the time-invariant covariates that affect the transaction process.
+//' Each column represents a different covariate. For every column a gamma parameter
+//' needs to added to \code{vCovParams_trans} at the respective position.
+//'
+//' \code{mCov_life} is a matrix containing the covariates data of
+//' the time-invariant covariates that affect the lifetime process.
+//' Each column represents a different covariate. For every column a gamma parameter
+//' needs to added to \code{vCovParams_life} at the respective position.
```

Comment:
these really need to be move to templates. re-writting them in every function is really silly (also for pnbd....)
This is afaik only possible since recently because the template needs to contain the `@details` marker as well. 
ie 
```
@details mCov_life  blabal
@details mCov_trans blabla
```